test/integration/externalnode_test.go

```
@@ -0,0 +1,361 @@
+		if len(stringList) > 0 {
+			actualResult = stringList[0]
+		} else {
+			actualResult = ""
+		}
+		if agented == true {
+			logf.Log.Info("Validate External Node Name", "Actual", actualResult, "Expected", expectedResult)
+			Expect(actualResult == expectedResult).To(BeTrue())
+
+			// Validate spec field in the generated External Node CR.
+			spec := &antreav1alpha1.ExternalNodeSpec{
+				Interfaces: []antreav1alpha1.NetworkInterface{
+					{
+						Name: strings.ToLower(nic),
+						IPs:  []string{privateIp, publicIp},
+					},
+				},
+			}
+			logf.Log.Info("Validate External Node Spec", "Actual", &en.Spec, "Expected", spec)
+			Expect(&en.Spec).To(Equal(spec))
+		} else {
+			logf.Log.Info("Validate External Entity Name", "Actual", actualResult, "Expected", expectedResult)
+			Expect(actualResult == expectedResult).To(BeTrue())
+
+			// Validate spec field in the generated External Entity CR.
+			spec := &antreav1alpha2.ExternalEntitySpec{
+				Endpoints: []antreav1alpha2.Endpoint{
+					{
+						IP: privateIp,
+					},
+					{
+						IP: publicIp,
+					},
+				},
+				ExternalNode: config.ANPNepheController,
+			}
+			logf.Log.Info("Validate External Entity Spec", "Actual", &ee.Spec, "Expected", spec)
+			Expect(&ee.Spec).To(Equal(spec))
+		}
+	}
+
+	checkEnCreated := func(key client.ObjectKey, en *antreav1alpha1.ExternalNode) error {
+		logf.Log.Info("Check for External Node creation")
+		err := wait.Poll(time.Second*5, time.Second*60, func() (bool, error) {
+			ret := k8sClient.Get(context.TODO(), key, en)
+			if ret != nil {
+				logf.Log.Error(ret, "External Node fetch failed", "fetch key", key)
+				return false, ret
+			}
+			return true, nil
+		})
+		return err
+	}
+
+	checkEeCreated := func(key client.ObjectKey, ee *antreav1alpha2.ExternalEntity) error {
+		logf.Log.Info("Check for External Entity creation")
+		err := wait.Poll(time.Second*5, time.Second*60, func() (bool, error) {
+			ret := k8sClient.Get(context.TODO(), key, ee)
```

if the object is not present, does it return error. if so, we will be logging error on every call. Verify and avoid 192 logging. have a final check on 197 to say failure

---

Returning error from function and logging on the caller on error. Also, made sure that failure is not returned on poll failure within the timeout